sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -417,6 +450,22 @@ public Builder addArray(Object ... values) {
       return verifiedList;
     }
 
+    private Map<Object, Object> verifyMap(Object value, FieldType componentKeyType,
+        FieldType componentValueType, String fieldName) {
+      if (!(value instanceof Map)) {
+        throw new IllegalArgumentException(String.format(
+            "For field name %s and map type expected Map class. Instead " + "class type was %s.",
+            fieldName, value.getClass()));
+      }
+      Map<Object, Object> valueMap = (Map<Object, Object>) value;
+      Map<Object, Object> verifiedMap = Maps.newHashMapWithExpectedSize(valueMap.size());
+      for (Entry<Object, Object> kv : valueMap.entrySet()) {
+        verifiedMap.put(verifyPrimitiveType(kv.getKey(), componentKeyType.getTypeName(), fieldName),
```

Comment:
<!--new_thread; commit:7bf9d2048e8d568ee9280629db391775cc490358; resolved:0-->
This is actually incorrect right now, since as coded the key type might not be a primitive type. However as I mentioned in Schema.java, I think it would be better to change the key type to be a TypeName, in which case this logic will be correct.